### PR TITLE
remove vendor directory

### PR DESCRIPTION
**Pull request by jacobbednarz**
This is no longer needed because we have the `composer.lock` file
committed and the CI/release processes all utilise `composer install`
with caching (including the SVN repository generation).

Do not merge until #358 is merged.

### CODE REVIEW

**Review by patryk — APPROVED** [review]

**Comment by manatarms** [comment]
As long as we test a clean composer install and validate everything works, LGTM.

**Review by manatarms — APPROVED** [review]

**Review by mgirouard — APPROVED** [review]

**Comment by jacobbednarz** [comment]
Will need to be updated following the changes in #363.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,13 +14,7 @@ jobs:
...
       with:
         command: install
-        version: 1
-        php_version: 7.4
-
-    - name: Remove intentional crash test in dependency
-      uses: php-actions/composer@v5
-      with:
-        command: run-script clean-crash-test
+        dev: no
```
Comment (jacobbednarz) [inline .github/workflows/release.yml:17]: `--no-dev` for releases

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 *.DS_Store
+vendor/
```
Comment (jacobbednarz) [inline .gitignore:2]: yay! don't allow people to commit this directory anymore 🎉

**Review by mgirouard — APPROVED** [review]

**Pull request merged**